homework_2/internal/infrastructure/http/axilisOfferFeedCSV.go

```diff
@@ -0,0 +1,108 @@
+package http
+
+import (
+	"code-cadets-2021/homework_2/internal/domain/models"
+	"context"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const axilisFeedCSVURL = "http://18.193.121.232/axilis-feed-2"
+
+type AxilisOfferFeedCSV struct {
+	httpClient http.Client
+	updates    chan models.Odd
+}
+
+func NewAxilisOfferFeedCSV(httpClient http.Client) *AxilisOfferFeedCSV {
+	return &AxilisOfferFeedCSV{
+		httpClient: httpClient,
+		updates:    make(chan models.Odd),
+	}
+}
+
+func (a *AxilisOfferFeedCSV) Start(ctx context.Context) error {
+	// repeatedly:
+	// - get odds from HTTP server
+	// - write them to updates channel
+	// - if context is finished, exit and close updates channel
+	// (test your program from cmd/main.go)
+
+	for {
+		content, err := a.getOddsFromServer()
+		if err != nil {
+			return err
+		}
+
+		select {
+		case <-ctx.Done():
+			close(a.updates)
+			return nil
```

Comment:
We usually put `<-ctx.Done()` as the last parameter in select-case, but the way you did it is also fine.